doc/release/upcoming_changes/25531.improvement.rst

```diff
@@ -0,0 +1,6 @@
+Guard PyArrayScalar_VAL and PyUnicodeScalarObject for the limited API
```

Comment:
Could add doublel ticks evne in the title, but doesn't matter.

Reply:
Maybe @charris has an opinion?

Reply:
I think either way is fine. I don't worry too much about those details, but do change the single backticks to double bacticks in the titles because they are included in the `*.rst` release notes.